### ISSUE TITLE
fail to upgrade scylla smoothly (without clean repos cache)

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#2822]
*Installation details*
Scylla version (or git commit hash):  `scylla-1.7.5-0.20170918.760af56.el7.centos.x86_64` -> `scylla-2.0.rc5-0.20170917.0b49cfc.el7.centos.x86_64.rpm`
Cluster size: 3
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS

*Description*
Try to upgrade scylla from 1.7.5 to 2.0.rc5. But I can't update scylla directly by `sudo yum update scylla\*` after update repo to [2.0.rc5 repo](http://downloads.scylladb.com.s3.amazonaws.com/rpm/unstable/centos/branch-2.0/24/scylla.repo).
[Old repo link](http://downloads.scylladb.com.s3.amazonaws.com/rpm/centos/scylla-1.7.repo)

```
[scylla-test@gce-refresh-1-7-amos-db-node-228ee1c4-0-1 ~]$ sudo cp scylla.repo /etc/yum.repos.d/
[scylla-test@gce-refresh-1-7-amos-db-node-228ee1c4-0-1 ~]$ sudo yum update scylla\*
Loaded plugins: fastestmirror
scylla                                                                                                  | 3.3 kB  00:00:00     
Not using downloaded scylla/repomd.xml because it is older than what we have:
  Current   : Mon Sep 18 13:12:03 2017
  Downloaded: Mon Sep 18 02:33:20 2017
scylla-3rdparty                                                                                         | 3.4 kB  00:00:00     
Not using downloaded scylla-3rdparty/repomd.xml because it is older than what we have:
  Current   : Mon Sep 18 13:12:13 2017
  Downloaded: Mon Sep 18 02:33:27 2017
scylla-3rdparty-generic                                                                                 | 3.3 kB  00:00:00     
Not using downloaded scylla-3rdparty-generic/repomd.xml because it is older than what we have:
  Current   : Mon Sep 18 13:12:16 2017
  Downloaded: Mon Sep 18 02:33:30 2017
scylla-generic                                                                                          | 3.3 kB  00:00:00     
Not using downloaded scylla-generic/repomd.xml because it is older than what we have:
  Current   : Mon Sep 18 13:12:10 2017
  Downloaded: Mon Sep 18 02:33:24 2017
Loading mirror speeds from cached hostfile
 * base: mirrors.liquidweb.com
 * epel: mirror.us.leaseweb.net
 * extras: mirror.teklinks.com
 * updates: www.gtlib.gatech.edu
No packages marked for update
[scylla-test@gce-refresh-1-7-amos-db-node-228ee1c4-0-1 ~]$
```

Workaround:
clean cache before upgrade by `sudo yum clean all`

Before upgrade:
```
[scylla-test@gce-refresh-1-7-amos-db-node-228ee1c4-0-1 ~]$ rpm -qa |grep scylla 
scylla-env-1.0-1.el7.centos.noarch
scylla-server-1.7.5-0.20170918.760af56.el7.centos.x86_64
scylla-tools-1.7.5-20170918.bc88754.el7.centos.noarch
scylla-gdb-7.10.1-30.el7.centos.x86_64
scylla-conf-1.7.5-0.20170918.760af56.el7.centos.x86_64
scylla-jmx-1.7.5-20170918.1c249f9.el7.centos.noarch
scylla-debuginfo-1.7.5-0.20170918.760af56.el7.centos.x86_64
scylla-1.7.5-0.20170918.760af56.el7.centos.x86_64
scylla-kernel-conf-1.7.5-0.20170918.760af56.el7.centos.x86_64
```

**Comment by tzach** [comment]
@amoskong @syuu1228 if no better solution we need to add this bypass to the upgrade doc

**Comment by syuu1228** [comment]
@tzach @amoskong Okay I'll try to working on this now

**Comment by syuu1228** [comment]
@tzach @amoskong I couldn't reproduce this while testing on my local docker environment and AWS.
I think repository itself is OK.

The error message @amoskong reported says you are trying to overwrite same repo name "[scylla]" by installing new .repo file, but actually newly specified repo contains older metadata so yum command ignore the repository.

Since 2.0rc5 is not released so repository name on [the .repo file](http://downloads.scylladb.com.s3.amazonaws.com/rpm/unstable/centos/branch-2.0/24/scylla.repo) is 
 "[scylla]" style repo name, not "[scylla-2.0]" style repo name. That's fine, but where previous '[scylla]" repo comes from?

@amoskong , is this [the 1.7 .repo file](http://downloads.scylladb.com.s3.amazonaws.com/rpm/centos/scylla-1.7.repo) you used in the test?
It used "[scylla-1.7]", not "[scylla]".
Have you tested other versions of Scylla, by using other .repo file?

Not sure which one, but yum command talking about some repo witch is newer than 2.0.rc5, and it's probably nightly build (since we uses [scylla-X.Y] style repo name on released branch).

Anyway, this problem will not occurs on users side. Because we don't release under "[scylla]" stlye repo name.

**Comment by amoskong** [comment]
@syuu1228 In 'released' [scylla-2.0.repo](http://downloads.scylladb.com.s3.amazonaws.com/rpm/centos/scylla-2.0.repo), it used `[scylla*-2.0]`, this problem doesn't exist.
When I reported the issue, I used [unstable 2.0.rc5 repo](http://downloads.scylladb.com.s3.amazonaws.com/rpm/unstable/centos/branch-2.0/24/scylla.repo).

Sure, user won't touch this problem. So we don't need to update upgrade doc. @tzach 
Let's close this issue. Thanks!
